.gemspec

```diff
@@ -1,36 +1,72 @@
 # -*- encoding: utf-8 -*-
 #
 # PLEASE NOTE
+# TODO: Add a comment here about building the gem using the "PLATFORM"
+# environment variable.
+#
 # This gemspec is not intended to be used for building the Puppet gem.  This
 # gemspec is intended for use with bundler when Puppet is a dependency of
 # another project.  For example, the stdlib project is able to integrate with
 # the master branch of Puppet by using a Gemfile path of
 # git://github.com/puppetlabs/puppet.git
-#
-# Please see the [packaging
-# repository](https://github.com/puppetlabs/packaging) for information on how
-# to build the Puppet gem package.
+
+PROJECT_ROOT = File.join(File.dirname(__FILE__))
 
 Gem::Specification.new do |s|
   s.name = "puppet"
-  version = "5.5.0"
-  mdata = version.match(/(\d+\.\d+\.\d+)/)
-  s.version = mdata ? mdata[1] : version
+
+  ## READ THE GEM VERSION ##
+  version_file = File.join(PROJECT_ROOT, 'lib', 'puppet', 'version.rb')
+  version_line_regex = /[^\s]*VERSION = ['"](\d+\.\d+\.\d+)['"]/
+  File.open(version_file).each_line(version_file) do |line|
```

Comment:
I think it might be cleaner to use an instance_eval on the version file and actually read `Puppet.version` instead of searching for the string.

Reply:
There might be pitfalls with that I'm unaware of, just a thought

Reply:
Thanks for the suggestion! I cleaned it up to just read the entire file and match on the contents. Should be easier to read now, hopefully.

I really don't like instance_eval because that loads an entire module for no reason, plus it did not work when I used it.